### ISSUE TITLE
snake_case Parameters seem to be converted unnecessarily to camelCase

### ISSUE THREAD

**Issue by komali2** [kubb-labs/kubb#1395]
### What version of `kubb` is running?

3.05

### What platform is your computer?

linux

### What version of external packages are you using(`@tanstack-query`, `MSW`, `React`, `Vue`, ...)

tanstack query: 5.51.23, react: 18, 

### What steps can reproduce the bug?

Integrate this into an `openapi.json` file:

```
    "/steps/{step_id}": {
      "delete": {
        "tags": [
          "Steps"
        ],
        "summary": "Flag a step as deleted",
        "operationId": "delete_step",
        "security": [
          {
            "OAuth2PasswordBearer": []
          }
        ],
        "parameters": [
          {
            "name": "step_id",
            "in": "path",
            "required": true,
            "schema": {
              "type": "integer",
              "exclusiveMinimum": 0,
              "title": "Step Id"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Step"
                }
              }
            }
          },
          "400": {
            "description": "Bad Request",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "errors": {
                      "type": "array",
                      "items": {
                        "type": "object",
                        "properties": {
                          "code": {
                            "type": "string"
                          },
                          "field": {
                            "type": "array",
                            "items": {
                              "type": "string"
                            }
                          },
                          "message": {
                            "type": "string"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          "401": {
            "description": "Unauthorized",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "403": {
            "description": "Forbidden",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "404": {
            "description": "Not Found",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      },
      "get": {
        "tags": [
          "Steps"
        ],
        "summary": "Get a step",
        "operationId": "get_step",
        "security": [
          {
            "OAuth2PasswordBearer": []
          }
        ],
        "parameters": [
          {
            "name": "step_id",
            "in": "path",
            "required": true,
            "schema": {
              "type": "integer",
              "exclusiveMinimum": 0,
              "title": "Step Id"
            }
          }
        ],
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Step"
                }
              }
            }
          },
          "400": {
            "description": "Bad Request",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "errors": {
                      "type": "array",
                      "items": {
                        "type": "object",
                        "properties": {
                          "code": {
                            "type": "string"
                          },
                          "field": {
                            "type": "array",
                            "items": {
                              "type": "string"
                            }
                          },
                          "message": {
                            "type": "string"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          "401": {
            "description": "Unauthorized",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "403": {
            "description": "Forbidden",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "404": {
            "description": "Not Found",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      },
      "patch": {
        "tags": [
          "Steps"
        ],
        "summary": "Update a step",
        "operationId": "update_step",
        "security": [
          {
            "OAuth2PasswordBearer": []
          }
        ],
        "parameters": [
          {
            "name": "step_id",
            "in": "path",
            "required": true,
            "schema": {
              "type": "integer",
              "exclusiveMinimum": 0,
              "title": "Step Id"
            }
          }
        ],
        "requestBody": {
          "required": true,
          "content": {
            "application/json": {
              "schema": {
                "$ref": "#/components/schemas/UpdateStep"
              }
            }
          }
        },
        "responses": {
          "200": {
            "description": "Successful Response",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/Step"
                }
              }
            }
          },
          "422": {
            "description": "Unprocessable Entity",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "code": {
                      "type": "string"
                    },
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "400": {
            "description": "Bad Request",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "errors": {
                      "type": "array",
                      "items": {
                        "type": "object",
                        "properties": {
                          "code": {
                            "type": "string"
                          },
                          "field": {
                            "type": "array",
                            "items": {
                              "type": "string"
                            }
                          },
                          "message": {
                            "type": "string"
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          },
          "401": {
            "description": "Unauthorized",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "403": {
            "description": "Forbidden",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "404": {
            "description": "Not Found",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          },
          "409": {
            "description": "Conflict",
            "content": {
              "application/json": {
                "schema": {
                  "type": "object",
                  "properties": {
                    "code": {
                      "type": "string"
                    },
                    "field": {
                      "type": "string"
                    },
                    "message": {
                      "type": "string"
                    }
                  }
                }
              }
            }
          }
        }
      }
    }
```

Follow a config similar to this:

```
import { defineConfig } from "@kubb/core";
import { pluginTs } from "@kubb/plugin-ts";
import { pluginZod } from "@kubb/plugin-zod";
import { pluginReactQuery } from "@kubb/plugin-react-query";
import { pluginOas } from "@kubb/plugin-oas";

export default defineConfig({
  root: ".",
  input: {
    path: "./apps/backend/openapi.json",
  },
  output: {
    path: "./apps/webapp/src/gen",
    clean: true,
  },
  hooks: {
    done: ["npm run lintfix", "npm run format"],
  },
  plugins: [
    pluginOas(),
    pluginTs({
      output: {
        path: "models",
      },
      group: {
        type: "tag",
        name: ({ group }) => `${group}Controller`,
      },
      enumType: "literal",
    }),
    pluginZod({
      output: {
        path: "./zod",
      },
      group: {
        type: "tag",
        name: ({ group }) => `${group}Schemas`,
      },
      typed: true,
      dateType: "stringOffset",
      unknownType: "unknown",
      importPath: "zod",
    }),
    pluginReactQuery({
      output: {
        path: "./hooks",
      },
      group: {
        type: "tag",
        name: ({ group }) => `${group}Hooks`,
      },
      client: {
        dataReturnType: "full",
      },
      mutation: {
        methods: ["post", "put", "delete"],
      },
      infinite: {
        queryParam: "page",
        initialPageParam: 0,
        cursorParam: undefined,
      },
      query: {
        methods: ["get"],
        importPath: "@tanstack/react-query",
      },
      parser: "zod",
      pathParamsType: "object",
    }),
  ],
});

```

Generate. 

Observe that `step_id` is converted, erroneously and in such a way as to lead to a typeerror, `stepId`:

```
import client from "@kubb/plugin-client/client";
import type {
  GetStepQueryResponse,
  GetStepPathParams,
  GetStep400,
  GetStep401,
  GetStep403,
  GetStep404,
} from "../../models/stepsController/GetStep.ts";
import type { RequestConfig, ResponseConfig } from "@kubb/plugin-client/client";
import type {
  QueryKey,
  QueryObserverOptions,
  UseQueryResult,
} from "@tanstack/react-query";
import { getStepQueryResponseSchema } from "../../zod/stepsSchemas/getStepSchema.ts";
import { queryOptions, useQuery } from "@tanstack/react-query";

export const getStepQueryKey = ({
  stepId,
}: {
  stepId: GetStepPathParams["step_id"];
}) => [{ url: "/steps/:step_id", params: { stepId: stepId } }] as const;

export type GetStepQueryKey = ReturnType<typeof getStepQueryKey>;

/**
 * @summary Get a step
 * @link /steps/:step_id
 */
async function getStep(
  { stepId }: GetStepPathParams,
  config: Partial<RequestConfig> = {}
) {
  const res = await client<
    GetStepQueryResponse,
    GetStep400 | GetStep401 | GetStep403 | GetStep404,
    unknown
  >({ method: "GET", url: `/steps/${stepId}`, ...config });
  return { ...res, data: getStepQueryResponseSchema.parse(res.data) };
}

export function getStepQueryOptions(
  { stepId }: GetStepPathParams,
  config: Partial<RequestConfig> = {}
) {
  const queryKey = getStepQueryKey({ stepId });
  return queryOptions({
    enabled: !!stepId,
    queryKey,
    queryFn: async ({ signal }) => {
      config.signal = signal;
      return getStep({ stepId }, config);
    },
  });
}

```

### How often does this bug happen?

Every time

### What is the expected behavior?

Parameters should maintain the casing provided to them. 

### Swagger/OpenAPI file?

Attached above. 

### Additional information

Love this software, a complete game changer, can't wait to get it integrated!

**Comment by stijnvanhulle** [comment]
When using js/ts you cannot use variables that contain special char(- and _ for example). That is why we change all variables to pascalCase but keep the type as it was defined in the swagger/openApi file. 

This means that the generated files will use another format but they will call the same api endpoints as defined in your swagger/OpenApi file. I will open a pr so that the type will be correct bind.

**Comment by stijnvanhulle** [comment]
`v3.0.6` will resolve your issue.

**Comment by komali2** [comment]
Thank you very much for your help. 

I'm curious about your point regarding "special characters." Indeed, you can't use the character `-` in variable names in javascript or typescript:

```
const some-variable='hello2'
```

Will result in a `SyntaxError`. 

However, to the best of my understanding, and per my testing and experience, `_` is a valid character at any location in a javascript or typescript variable name:

```
const some_variable = 'hello';
console.log(some_variable);
```

Further info: https://stackoverflow.com/questions/1661197/what-characters-are-valid-for-javascript-variable-names

In particular: https://stackoverflow.com/a/3155352/2590119

We personally would like to use the same style on the frontend as we do on the backend, which uses `snake_case`, hence the issue I raised.

**Comment by stijnvanhulle** [comment]
Fixed with v`3.0.11`.

Now all parameters will not be changed to camelcase, so in your example it will do a check if it can be valid js/ts. If that is not the case, it will fallback on transforming the name with our camelcase/pascalcase helper.

**Comment by komali2** [comment]
confirmed fixed! thank you very much
